tests/device_tests/test_protection_levels.py

```diff
@@ -32,239 +33,362 @@
+                label="label",
+                language="en-US",
             )
+        )
 
-    @pytest.mark.setup_client(pin=PIN4, passphrase=True)
-    def test_sign_message(self, client):
-        with client:
-            client.use_pin_sequence([PIN4])
-            client.set_expected_responses(
-                [
-                    messages.ButtonRequest,
-                    messages.PinMatrixRequest,
-                    messages.PassphraseRequest,
-                    messages.MessageSignature,
-                ]
-            )
-            btc.sign_message(client, "Bitcoin", [], "testing message")
 
-    @pytest.mark.setup_client(pin=PIN4, passphrase=True)
-    def test_verify_message(self, client):
-        with client:
-            client.set_expected_responses(
-                [messages.ButtonRequest, messages.ButtonRequest, messages.Success]
-            )
-            btc.verify_message(
-                client,
-                "Bitcoin",
-                "14LmW5k4ssUrtbAB4255zdqv3b4w1TuX9e",
-                bytes.fromhex(
-                    "209e23edf0e4e47ff1dec27f32cd78c50e74ef018ee8a6adf35ae17c7a9b0dd96f48b493fd7dbab03efb6f439c6383c9523b3bbc5f1a7d158a6af90ab154e9be80"
-                ),
-                "This is an example of a signed message.",
+@pytest.mark.setup_client(uninitialized=True)
+@pytest.mark.skip_t2
+def test_recovery_device(client):
+    client.use_mnemonic(MNEMONIC12)
+    with client:
+        assert client.features.pin_protection is False
+        client.set_expected_responses(
+            [messages.ButtonRequest]
+            + [messages.WordRequest] * 24
+            + [messages.Success, messages.Features]
+        )
+
+        device.recover(
+            client, 12, False, False, "label", "en-US", client.mnemonic_callback
+        )
+        assert client.features.passphrase_protection is False
+
+    with pytest.raises(TrezorFailure):
+        # This must fail, because device is already initialized
+        # Using direct call because `device.recover` has its own check
+        client.call(
+            messages.RecoveryDevice(
+                word_count=12,
+                passphrase_protection=False,
+                pin_protection=False,
+                label="label",
+                language="en-US",
             )
+        )
+
 
-    @pytest.mark.setup_client(pin=PIN4, passphrase=True)
-    def test_signtx(self, client):
-        inp1 = messages.TxInputType(
-            address_n=[0],  # 14LmW5k4ssUrtbAB4255zdqv3b4w1TuX9e
-            prev_hash=TXHASH_d5f65e,
-            prev_index=0,
-            amount=390000,
+def test_sign_message(client):
+    # Even the order of responses is different here
+    if client.features.model == "1":
+        expected_responses = [
+            messages.ButtonRequest,
+            messages.PinMatrixRequest,
+            messages.PassphraseRequest,
+            messages.MessageSignature,
+        ]
+    else:
+        expected_responses = [
```

Comment:
please split into two tests.

please mark which of the ButtonRequests is the PIN request -- either by using _pin_request() or by specifying a code

Reply:
Fixed in 699ae28